File: 16.py
#翻转链表
class ListNode(object):
	"""docstring for ListNode"""
	def __init__(self, value):
		#super(ListNode, self).__init__()
		self.next = None
		self.value = value
def reverse_list(node):
	p_head = None
	p_node = node
	p_pre = None
	while p_node:
		p_next = p_node.next
		if p_next == None:
			p_head = p_node
		p_node.next = p_pre
		p_pre = p_node
		p_node = p_next
	return p_head

node1 = ListNode(10)
node2 = ListNode(11)
node3 = ListNode(13)
node1.next = node2
node2.next = node3
p = reverse_list(node1)
print(p.value)

File: 16.cpp
#include <iostream>
using namespace std;
struct ListNode
{
	int value;
	ListNode* next;
};
ListNode* reverseNode(ListNode* head)
{
	//Node为NULL
	ListNode* pNode = NULL;
	if(head == NULL){
		return NULL;
	}
	else if(!head->next)
	{

		return head;
	}else{

		pNode = head;
		ListNode* pNext = pNode;
		ListNode* pPre = NULL;
		while(pNode->next!=NULL){
		//	cout<<pNode->value;
			pNext = pNode->next;
			pNode->next = pPre;
			pPre = pNode;
			pNode = pNext;		
		}
		return pNode;
	}
	
	//只有一个Node，返回该Node

	
}
void main(){
	ListNode* a = new ListNode();
	ListNode* b = new ListNode();
	ListNode* c = new ListNode();
	ListNode* d = new ListNode();
	ListNode* e = new ListNode();
	a->next = b;
	b->next = c;
	c->next = d;
	d->next = e;
	d->next = NULL;
	a->value =1;
	b->value =2;
	c->value =3;
	d->value =4;
	ListNode* h= new ListNode();
	h=NULL;
    ListNode* f = new ListNode();
    f->value =10;
	//f->next=NULL;
    //ListNode* ph = reverseNode(f);
	cout<<(!f->next);
	int stop;
	cin>>stop;
}